Rubberduck.Core/UI/CodeExplorer/Commands/IndentCommand.cs

```diff
@@ -1,127 +1,117 @@
+using System;
+using System.Collections.Generic;
 using Rubberduck.Navigation.CodeExplorer;
 using Rubberduck.Parsing.VBA;
 using Rubberduck.SmartIndenter;
-using Rubberduck.UI.Command;
 using System.Linq;
-using NLog;
 using Rubberduck.Parsing.Annotations;
 using Rubberduck.Parsing.Symbols;
 using Rubberduck.Interaction.Navigation;
 
 namespace Rubberduck.UI.CodeExplorer.Commands
 {
-    public class IndentCommand : CommandBase
+    public class IndentCommand : CodeExplorerCommandBase
     {
+        private static readonly Type[] ApplicableNodes =
+        {
+            typeof(CodeExplorerProjectViewModel),
+            typeof(CodeExplorerCustomFolderViewModel),
+            typeof(CodeExplorerComponentViewModel),
+            typeof(CodeExplorerMemberViewModel)
```

Comment:
Just a thought - this seems to appear in quite a few of the commands. Why not have those types inherit from a 2nd abstract class or an interface to indicate that they are a project object for which command can apply? 

Alternatively provide an extension class that abstracts the oft-used enumerations of types?